tfx/extensions/experimental/kfp_compatibility/kfp_container_component.py

```diff
@@ -0,0 +1,112 @@
+from tfx.types.experimental.simple_artifacts import File
+from typing import Any, Callable, Dict, Text
+
+
+def load_kfp_yaml_container_component(
+    path: Text
+) -> Callable[..., base_component.BaseComponent]:
+  """Creates a container-based component from a Kubeflow component spec.
+
+  See
+  https://www.kubeflow.org/docs/pipelines/reference/component-spec/
+
+  Example:
+    component = load_kfp_yaml_container_component(
+      "kfp_pipelines_root/components/datasets/Chicago_Taxi_Trips/component.yaml"
+    )
+
+  Args:
+    path: local file path of a Kubeflow Pipelines component
+                         YAML file.
+
+  Returns:
+    Container component that can be instantiated in a TFX pipeline.
+  """
+  with open(path) as component_file:
+    data = yaml.load(component_file, Loader=yaml.FullLoader)
+  convert_target_fields_to_kv_pair(data)
+  component_spec = json_format.ParseDict(
+      data, kfp_component_spec_pb2.ComponentSpec())
+  container = component_spec.implementation.container
+  command = (list(map(get_command_line_argument_type, container.command)) +
+             list(map(get_command_line_argument_type, container.args)))
+  # TODO(ericlege): Support classname to class translation in inputs.type
+  inputs = {item.name: File for item in component_spec.inputs}
+  outputs = {item.name: File for item in component_spec.outputs}
+  parameters = {}
+  return container_component.create_container_component(
+      name=component_spec.name,
+      image=container.image,
+      command=command,
+      inputs=inputs,
+      outputs=outputs,
+      parameters=parameters,
+    )
+
+
+def convert_target_fields_to_kv_pair(
+    parsed_dict: Dict[Text, Any]
+) -> None:
+  """ Converts in place specific string fields to key value pairs of {constantValue: [Text]} for proto3 compatibility.
+
+  Args:
+    parsed_dict: dictionary obtained from parsing a Kubeflow component spec.
+                 This argument is modified in place.
+
+  Returns:
+    None
+  """
+  conversion_string_paths = [
+      ['implementation', 'container', 'command'],
+      ['implementation', 'container', 'args'],
+  ]
+  for path in conversion_string_paths:
+    parsed_dict_location = parsed_dict
+    for label in path:
+      parsed_dict_location = parsed_dict_location.get(label, {})
+    if isinstance(parsed_dict_location, list):
+      for ind, value in enumerate(parsed_dict_location):
+        if isinstance(value, str):
+          parsed_dict_location[ind] = {"constantValue": value}
+
+
+def get_command_line_argument_type(
+    command: kfp_component_spec_pb2.StringOrPlaceholderer
+) -> executor_specs.CommandlineArgumentType:
+  """ Converts a container command to the corresponding type under executor_specs.CommandlineArgumentType.
+
+  Args:
+    command: StringOrPlaceholderer which encodes a container command.
+
+  Returns:
+    command to be passed into create_container_component.
+  """
+  if command.HasField("constantValue"):
```

Comment:
Are there plans to support `IfPlaceholder`? Many python-based components use it. (JFYI: It's compile-time conditional placeholder.)

Reply:
This is not supported in this first pass for now.

Reply:
/cc: @chensun @neuromage